### PR TITLE
Adds `suppress_warnings` option to configuration

### PR DESCRIPTION
**Pull request by HorlogeSkynet**
## Description

Simply redirects `stderr` to `/dev/null` (or set it back to `STDERR` in function of `suppress_warning` configuration option value.

## Reason and / or context

May be useful for some people 

## How has this been tested ?

On Debian Stretch, redirecting and setting back work like a charm

## Types of changes :

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Typo / style fix (non-breaking change which improves readability)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist :

- [X] My change looks good ;
- [X] I have updated the _README.md_ file accordingly ;
- [X] I agree that my code may be modified in the future ;
- [X] My code follows the code style of this project ([PEP8](https://www.python.org/dev/peps/pep-0008/)).

### CODE REVIEW

**Comment by AdJaGu** [comment]
I have tested this on my Arch Linux box using the config option `suppress_warnings` set to `true` and `false` and received no errors.

Is there another way I could test this? I ask this because when I run archey with or without this update I receive no errors. Would removing a dependency and re-testing be a good idea?

**Comment by HorlogeSkynet** [comment]
As you want, I've already run these tests, but a quick way to check this would be : `PATH='/tmp'; /usr/bin/python3 ./archey`

Btw, I think we really should have test cases on this project, but I have no idea how we could implement them...

**Comment by AdJaGu** [comment]
Running `PATH='/tmp'; /usr/bin/python3 ./archey4` with `suppress_warnings` set to `true` or `false` results in the following: `Please, install first 'procps' on your distribution.`

```
[doctor@who ~]$ pacman -Ss procps
core/procps-ng 3.3.12-2 (base) [installed]
    Utilities for monitoring your system and its processes
```

**Comment by HorlogeSkynet** [comment]
Yeah sorry my bad, actually this call is done before the configuration loading.
What do you think about moving `config.loadConfiguration()` calls just after `config = Configuration()` ?

Maybe it's time to move some instructions in a `main()` method, too ?

**Comment by AdJaGu** [comment]
I think it would be nice to have configuration options condensed into one block/section. That way the default configuration is in the same area that `config.loadConfiguration` is. As for test cases/testing, I have zero experience with that, but do think that tests/test cases would be useful.

**Comment by HorlogeSkynet** [comment]
I'm okay with that too, today I'll try to rework the "starting way" before the iteration over the entries...

For test cases, the principle would be to write some code importing Archey's classes, and testing each class output.
To do that, we need to "mock" execution dependencies (as `lsb_release`) as a dummy object to manipulate it.
On the long shot, I may do some research to implement this part...

**Comment by HorlogeSkynet** [comment]
LGTM now, keep me posted !
Gotta sleep a bit, bye 👋

**Comment by AdJaGu** [comment]
Upon testing the newest version of this branch I am now receiving `WAN_IP: No Address` and `archey` is running to fast. Total run time should take 6 seconds, but instead it takes 4 or 2 seconds.

Now if I run `./archey` multiple times in a row this is what happens.
Run 1: `WAN_IP:` returns `No Address` instead of populating `WAN_IP:`.
Run 2: `WAN_IP:` populates with proper external IPv4 and global IPv6.
Run 3: See Run 1.
Run 4: See Run 2.


Timing the running of `archey4` (that's what I named archey on my system when I installed it) almost always returns a `real` time of at least `0m5.XXXs` because I have set `ipv4_detection` and `ipv6_detection` to `5`. Results below.

Time archey4 run 1:
```
[doctor@who ~]$ time archey4
real 0m6.178s
user 0m0.326s
sys 0m0.194s
```
Time archey4 run 2:
```
[doctor@who ~]$ time archey4
real 0m6.674s
user 0m0.369s
sys 0m0.164s
```
Time archey4 run 3:
```
[doctor@who ~]$ time archey4
real 0m6.671s
user 0m0.336s
sys 0m0.200s
```
Time archey4 run 4:
```
[doctor@who ~]$ time archey4
real 0m6.663s
user 0m0.315s
sys 0m0.211s
```


Timing the running of `archey` (this branch) never reaches a `real` of `0m5.XXXs`, instead it is either a `real` of `0m4.XXXs` or `0m2.XXXs`. Results below.

Time archey run 1: `WAN_IP:` not populated.
```
[doctor@who hs_feat-suppress_warnings]$ time ./archey
real	0m4.471s
user	0m0.610s
sys	0m0.223s
```
Time archey run 2: `WAN_IP:` populated.
```
[doctor@who hs_feat-suppress_warnings]$ time ./archey
real	0m2.299s
user	0m0.460s
sys	0m0.299s
```
Time archey run 3: `WAN_IP:` not populated.
```
[doctor@who hs_feat-suppress_warnings]$ time ./archey
real	0m4.480s
user	0m0.598s
sys	0m0.240s
```
Time archey run 4: `WAN_IP:` populated.
```
[doctor@who hs_feat-suppress_warnings]$ time ./archey
real	0m2.552s
user	0m0.439s
sys	0m0.241s
```


Here is my `config.json` file:
```
{
	"allow_overriding": true,
	"suppress_warnings": false,
	"entries": {
		"User": true,
		"Hostname": true,
		"Model": true,
		"Distro": true,
		"Kernel": true,
		"Uptime": true,
		"WindowManager": true,
		"DesktopEnvironment": true,
		"Shell": true,
		"Terminal": true,
		"Packages": true,
		"Temperature": true,
		"CPU": true,
		"GPU": true,
		"RAM": true,
		"Disk": true,
		"LAN_IP": true,
		"WAN_IP": true
	},
	"colors_palette": {
		"use_unicode": true
	},
	"default_strings": {
		"no_address": "No Address",
		"not_detected": "Not detected"
	},
	"ip_settings": {
		"lan_ip_max_count": 2,
		"wan_ip_v6_support": true
	},
	"temperature": {
		"char_before_unit": " ",
		"use_fahrenheit": false
	},
	"timeout": {
		"ipv4_detection": 5,
		"ipv6_detection": 5
	}
}
```

TLDR;
`archey` (from this testing branch) does not seem to respect `timeout` settings `ipv4_detection` and `ipv6_detection` now and only returns a populated `WAN_IP:` on every second successive run.

`archey4` (from master branch, which is installed on my system) respects `timeout` settings and always populates `WAN_IP:`.

**Comment by HorlogeSkynet** [comment]
Okay so if I don't misunderstand, it looks like the configuration you set is not effectively loaded ?
In my case, each option is working correctly, so I guess you have an `allow_overriding` sets to `false` somewhere (or sets to `true` and **overridden** by the default one which have timeouts too short for your system).

Could you verify this (you can `print(self.config)` once all files have been loaded)  ?
Actually, I don't see another solution there, the logic didn't change very much about this 🤔

**Comment by AdJaGu** [comment]
You understood what I was saying and I was also wrong. After removing `config.json` from my `/home/doctor/github/hs_feat-suppress_warnings` folder `archey` ran just fine and respected/loaded my options set in `/home/doctor/.config/archey4/config.json`.

Everything looks good to me. I have no errors running this version of archey.

**Comment by HorlogeSkynet** [comment]
Perfect ! Can you accept the changes to allow the merge ? 🙂
